Formula/p/porter.rb

```diff
@@ -0,0 +1,27 @@
+class Porter < Formula
+  desc "App artifacts, tools, configs, and logic packaged as distributable installer"
+  homepage "https://porter.sh"
+  url "https://github.com/SpiffyEight77/porter.git",
+    tag:      "v1.0.18",
+    revision: "bcc69cc41dce4d85611dca7306848d56b7b1a4b1"
+  license "Apache-2.0"
+  head "https://github.com/getporter/porter.git", branch: "main"
+
+  depends_on "go" => :build
+  depends_on "mage" => :build
+
+  def install
+    system "mage", "-v", "BuildPorter"
+
+    bin.install "bin/porter"
+
+    generate_completions_from_executable(bin/"porter", "completion")
+
+    system "#{bin}/porter", "mixin", "install", "exec", "--version", "v#{version}"
+  end
+
+  test do
+    assert_match "Mixin commands. Mixins assist with authoring bundles.",
+                  shell_output("#{bin}/porter mixin -h")
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.